src/ws_correction.py

```diff
@@ -0,0 +1,183 @@
+                        rqa_moves[(i, plan.raw_field)] = ws_code_to_raw_field_map[ws_code.code_id]
+
+            # Build a dictionary of the demogs and follow-up survey fields that haven't been moved, and clear fields for those which have.
+            demogs_and_follow_up_survey_updates = dict()  # of raw_field -> updated value
+            for plan in PipelineConfiguration.DEMOGS_CODING_PLANS + PipelineConfiguration.FOLLOW_UP_CODING_PLANS:
+                if plan.raw_field in demogs_and_follow_up_survey_moves.keys():
+                    # Data is moving
+                    demogs_and_follow_up_survey_updates[plan.raw_field] = []
+                elif plan.raw_field in td:
+                    # Data is not moving
+                    demogs_and_follow_up_survey_updates[plan.raw_field] = [_WSUpdate(td[plan.raw_field], td[plan.time_field], plan.raw_field)]
+
+            # Build a list of the rqa fields that haven't been moved.
+            rqa_updates = []  # of (field, value)
+            for i, td in enumerate(group):
+                for plan in PipelineConfiguration.RQA_CODING_PLANS:
+                    if plan.raw_field in td:
+                        if (i, plan.raw_field) in rqa_moves.keys():
+                            # Data is moving
+                            pass
+                        else:
+                            # Data is not moving
+                            rqa_updates.append((plan.raw_field, _WSUpdate(td[plan.raw_field], td[plan.time_field], plan.raw_field)))
+
+            raw_demog_and_follow_up_survey_fields = {plan.raw_field for plan in PipelineConfiguration.DEMOGS_CODING_PLANS + \
+                                                     PipelineConfiguration.FOLLOW_UP_CODING_PLANS}
+            raw_rqa_fields = {plan.raw_field for plan in PipelineConfiguration.RQA_CODING_PLANS}
+
+            # Add data moving from demogs and follow-up survey fields to the relevant demog_/follow_up/rqa_updates
+            for plan in PipelineConfiguration.DEMOGS_CODING_PLANS + PipelineConfiguration.RQA_CODING_PLANS + \
+                        PipelineConfiguration.FOLLOW_UP_CODING_PLANS:
+                if plan.raw_field not in demogs_and_follow_up_survey_moves:
+                    continue
+                target_field = demogs_and_follow_up_survey_moves[plan.raw_field]
+                update = _WSUpdate(td[plan.raw_field], td[plan.time_field], plan.raw_field)
+                if target_field in raw_demog_and_follow_up_survey_fields:
+                    demogs_and_follow_up_survey_updates[target_field] = demogs_and_follow_up_survey_updates.get(target_field, []) + [update]
+                else:
+                    assert target_field in raw_rqa_fields, f"Raw field '{target_field}' not in any coding plan"
+                    rqa_updates.append((target_field, update))
+
+            # Add data moving from demog and follow up survey fields to the relevant demogs_/follow_up_/rqa_updates
```

Comment:
This comment is confusing, because it's the same as that for the previous block but is doing something different.